ros1_foxglove_bridge/include/foxglove_bridge/subscription.hpp

```diff
@@ -0,0 +1,43 @@
+#pragma once
+
+#include <functional>
+#include <memory>
+#include <mutex>
+#include <string>
+#include <string_view>
+
+#include <ros/node_handle.h>
+#include <ros/subscriber.h>
+
+namespace foxglove_bridge {
+
+class Subscription {
+public:
+  explicit Subscription(ros::NodeHandle& nh, const std::string& topic)
+      : nh_(nh)
+      , topic_(topic)
+      , num_subscribed_clients_(0)
```

Comment:
websocket_server.hpp already does subscription count tracking and only fires the subscribe/unsubscribe callbacks when subscription count goes from 0->1 or 1->0.

Reply:
Ah ok great, I have missed this!